aiven/provider.go

```diff
@@ -219,22 +220,30 @@ func generateClientTimeoutsSchema(timeouts map[string]time.Duration) *schema.Sch
 		Type:        schema.TypeSet,
 		MaxItems:    1,
 		Description: "Custom Terraform Client timeouts",
-		Optional:    true,
 		ForceNew:    true,
+		Optional:    true,
+		Deprecated:  "use timeouts instead",
 		Elem: &schema.Resource{
 			Schema: schemaTimeouts,
 		},
 	}
 }
 
 // getTimeoutHelper is a helper which extract from a resource data client timeout
+// Deprecated: getTimeoutHelper exists for historical compatibility
+// and should not be used. To set timeouts use native TF timeouts functionality.
 func getTimeoutHelper(d *schema.ResourceData, name string, defaultDuration time.Duration) (time.Duration, error) {
```

Comment:
Should defaultDuration be dropped and zero returned instead if a particular timeout isn't defined. Right now if you specify "client_timeout" for "read" and "timeout" for "update" the default value will be used for "update" instead of what was set in the config. That's of course a bit unlikely config but still doesn't seem like there's any benefit of defining default value here and there is some harm.

Reply:
I think this should do the trick since `client_timeout` has no default value in TF resource defenition:
```
clientTimeouts, ok := d.GetOk("client_timeout")
	if !ok || clientTimeouts.(*schema.Set).Len() == 0 {
		return 0, nil
	}```

But I think it is a good idea to remove `defaultDuration` for better readability and maybe I'm missing something.

Reply:
Did reread your commend - indeed you are right!